nodes/nomos-node-api/src/lib.rs

```diff
@@ -62,9 +61,14 @@ where
     }
 
     /// Service main loop
-    async fn run(mut self) -> Result<(), DynError> {
-        let endpoint = B::new(self.settings.backend_settings).await?;
-        endpoint.serve(self.handle).await?;
+    async fn run(mut self) -> Result<(), ServiceError> {
+        let endpoint = B::new(self.settings.backend_settings)
+            .await
+            .map_err(|e| ServiceError::Service(Box::new(e)))?;
+        endpoint
+            .serve(self.handle)
+            .await
+            .map_err(|e| ServiceError::Service(Box::new(e)))?;
```

Comment:
One thing that we can do is to `impl From DynErr for ServiceError`, that way we would not need to `map_err` constantly I think. Wdyt @al8n ? (this needs to be done in the overwatch branch).

Reply:
`From DynErr` not works so good, rust compiler is not smart enough, we have to use 
```rust
.map_err(|e| Box::new(e) as DynErr)?
```

Reply:
Uhm, let me take a shot at it.